File: backend/SWP391_FA25-binding/src/main/java/com/dangkhoa/superapp/swp391_fa25/entity/Dataset.java
package com.dangkhoa.superapp.swp391_fa25.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Table(name="Dataset")
@NoArgsConstructor
@Data
public class Dataset {
    @Id
    @GeneratedValue(strategy= GenerationType.IDENTITY)
    private int dataset_id;
    private int provider_id;
    private int port_id;
    private String name;
    private String description;
    private String category;
    private double price;
    private String access_type;
    private String data_format;
    private double data_size_mb;
    private LocalDateTime upload_date;
    private LocalDateTime last_updated;
    private String status;
    private String visibility;

    public Dataset(int provider_id, int port_id, String name, String description, String category, double price, String access_type, String data_format, double data_size_mb, LocalDateTime upload_date, LocalDateTime last_updated, String status, String visibility) {
        this.provider_id = provider_id;
        this.port_id = port_id;
        this.name = name;
        this.description = description;
        this.category = category;
        this.price = price;
        this.access_type = access_type;
        this.data_format = data_format;
        this.data_size_mb = data_size_mb;
        this.upload_date = upload_date;
        this.last_updated = last_updated;
        this.status = status;
        this.visibility = visibility;
    }

}

File: backend/EVDataMarketplace.API/Models/Dataset.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace EVDataMarketplace.API.Models;

[Table("Dataset")]
public class Dataset
{
    [Key]
    [Column("dataset_id")]
    public int DatasetId { get; set; }

    [Required]
    [Column("provider_id")]
    public int ProviderId { get; set; }

    [Column("port_id")]
    public int? PortId { get; set; }

    [Required]
    [MaxLength(255)]
    [Column("name")]
    public string Name { get; set; } = string.Empty;

    [Column("description")]
    public string? Description { get; set; }

    [MaxLength(150)]
    [Column("category")]
    public string? Category { get; set; }

    [MaxLength(50)]
    [Column("data_format")]
    public string? DataFormat { get; set; } = "CSV";

    [Column("row_count")]
    public int RowCount { get; set; } = 0; // Số dòng dữ liệu trong dataset

    [Column("upload_date")]
    public DateTime UploadDate { get; set; } = DateTime.Now;

    [Column("last_updated")]
    public DateTime? LastUpdated { get; set; }

    [MaxLength(50)]
    [Column("status")]
    public string? Status { get; set; } // Pending, Approved, Rejected, Active, Inactive

    [MaxLength(50)]
    [Column("visibility")]
    public string? Visibility { get; set; } // Public, Private

    [MaxLength(50)]
    [Column("moderation_status")]
    public string ModerationStatus { get; set; } = "Pending"; // Pending, UnderReview, Approved, Rejected

    // Navigation properties
    [ForeignKey("ProviderId")]
    public DataProvider? DataProvider { get; set; }

    public ICollection<DatasetModeration> DatasetModerations { get; set; } = new List<DatasetModeration>();
    public ICollection<DatasetRecord> DatasetRecords { get; set; } = new List<DatasetRecord>();
}
